lib/integration/esbuild.js

```diff
@@ -2,127 +2,206 @@
+  const remoteNs = name + '-remote'
+  const {extnames, process} = createFormatAwareProcessors(rest)
 
   return {name, setup}
 
   /**
    * @param {PluginBuild} build
    */
   function setup(build) {
-    build.onLoad({filter: extnamesToRegex(extnames)}, onload)
-  }
+    const filter = extnamesToRegex(extnames)
+    /* eslint-disable-next-line security/detect-non-literal-regexp */
+    const filterHttp = new RegExp('^https?:\\/{2}.+' + filter.source)
+    const filterHttpOrRelative = /^(https?:\/{2}|.{1,2}\/).*/
 
-  /**
-   * @param {Omit.<OnLoadArgs, 'pluginData'> & {pluginData?: {contents?: string}}} data
-   */
-  async function onload(data) {
-    /** @type {string} */
-    const doc =
-      data.pluginData && data.pluginData.contents !== undefined
-        ? data.pluginData.contents
-        : String(await fs.readFile(data.path))
-
-    let file = vfile({contents: doc, path: data.path})
-    /** @type {VFileMessage[]} */
-    let messages = []
-    /** @type {Message[]} */
-    const errors = []
-    /** @type {Message[]} */
-    const warnings = []
-    /** @type {VFileContents} */
-    let contents
-    /** @type {VFileMessage} */
-    let message
-    /** @type {Point} */
-    let start
-    /** @type {Point} */
-    let end
-    /** @type {number} */
-    let length
-    /** @type {number} */
-    let lineStart
-    /** @type {number} */
-    let lineEnd
-    /** @type {RegExpExecArray} */
-    let match
-    /** @type {number} */
-    let line
-    /** @type {number} */
-    let column
-
-    try {
-      file = await process(file)
-      contents = file.contents
-      messages = file.messages
-    } catch (error) {
-      error.fatal = true
-      messages.push(error)
+    if (allowDangerousRemoteMdx) {
+      // Intercept import paths starting with "http:" and "https:" so
+      // esbuild doesn't attempt to map them to a file system location.
+      // Tag them with the "http-url" namespace to associate them with
+      // this plugin.
+      build.onResolve(
+        {filter: filterHttp, namespace: 'file'},
+        resolveRemoteInLocal
+      )
+
+      build.onResolve(
+        {filter: filterHttpOrRelative, namespace: remoteNs},
+        resolveInRemote
+      )
     }
 
-    for (message of messages) {
-      start = message.location.start
-      end = message.location.end
-      length = 0
-      lineStart = 0
-      line = undefined
-      column = undefined
-
-      if (start.line != null && start.column != null && start.offset != null) {
-        line = start.line
-        column = start.column - 1
-        lineStart = start.offset - column
-        length = 1
-
-        if (end.line != null && end.column != null && end.offset != null) {
-          length = end.offset - start.offset
-        }
+    build.onLoad({filter: /.*/, namespace: remoteNs}, onloadremote)
+    build.onLoad({filter}, onload)
+
+    /** @param {OnResolveArgs} args  */
+    function resolveRemoteInLocal(args) {
```

Comment:
would:

```suggestion
    function resolveRemoteInLocal({path}) {
```

work here?